src/fingerprinting/hooks.ts

```diff
@@ -0,0 +1,88 @@
+import { Fingerprint, FingerprintInjector } from 'fingerprint-injector';
+import { BrowserPool, PlaywrightPlugin, PuppeteerPlugin } from '..';
+import { BrowserController } from '../abstract-classes/browser-controller';
+import { LaunchContext } from '../launch-context';
+import { getGeneratorDefaultOptions } from './utils';
+
+// @ts-expect-error I need help how to fix this :/
+export const createFingerprintPreLaunchHook = (browserPool: BrowserPool) => {
+    const {
+        fingerprintGenerator,
+        fingerprintCache,
+        fingerprintsOptions: {
+            fingerprintGeneratorOptions,
+        },
+    } = browserPool;
+
+    return (_pageId: string, launchContext: LaunchContext) => {
+        const { useIncognitoPages, launchOptions, proxyUrl } = launchContext;
+        // If no options are passed we try to pass best default options as possible to match browser and OS.
+        const fingerprintGeneratorFinalOptions = fingerprintGeneratorOptions || getGeneratorDefaultOptions(launchContext);
+        let fingerprint;
+
+        if (proxyUrl && fingerprintCache?.has(proxyUrl)) {
+            fingerprint = fingerprintCache.get(proxyUrl);
+        } else {
+            fingerprint = fingerprintGenerator.getFingerprint(fingerprintGeneratorFinalOptions).fingerprint;
+            fingerprintCache?.set(proxyUrl, fingerprint);
+        }
+
+        launchContext.extend({ fingerprint });
+
+        if (useIncognitoPages) {
+            return;
+        }
+        const { userAgent, screen } = fingerprint;
+        // @ts-expect-error I have no idea why I cannot assign property to unknown object.
```

Comment:
because unknown is not an object, it is now known what it is, hence the name _unknown_ :] 

this is all due to the generics on `BrowserPool`, keep it as it is for now, i will try to look into this once we merge the PR and simplify the typings so we can actually leverage them instead of ignoring :D